agents/addon/extension/elevenlabs_tts/elevenlabs_tts_extension.go

```diff
@@ -0,0 +1,347 @@
+		slog.Warn(fmt.Sprintf("GetProperty optional %s failed, err: %v", propertyStability, err), logTag)
+	} else {
+		elevenlabsTTSConfig.Stability = float32(stability)
+	}
+
+	if style, err := rte.GetPropertyFloat64(propertyStyle); err != nil {
+		slog.Warn(fmt.Sprintf("GetProperty optional %s failed, err: %v", propertyStyle, err), logTag)
+	} else {
+		elevenlabsTTSConfig.Style = float32(style)
+	}
+
+	if voiceId, err := rte.GetPropertyString(propertyVoiceId); err != nil {
+		slog.Warn(fmt.Sprintf("GetProperty optional %s failed, err: %v", propertyVoiceId, err), logTag)
+	} else {
+		if len(voiceId) > 0 {
+			elevenlabsTTSConfig.VoiceId = voiceId
+		}
+	}
+
+	// create elevenlabsTTS instance
+	elevenlabsTTS, err := newElevenlabsTTS(elevenlabsTTSConfig)
+	if err != nil {
+		slog.Error(fmt.Sprintf("newElevenlabsTTS failed, err: %v", err), logTag)
+		return
+	}
+
+	slog.Info(fmt.Sprintf("newElevenlabsTTS succeed with ModelId: %s, VoiceId: %s",
+		elevenlabsTTSConfig.ModelId, elevenlabsTTSConfig.VoiceId), logTag)
+
+	// set elevenlabsTTS instance
+	e.elevenlabsTTS = elevenlabsTTS
+
+	// pcm parameters
+	sampleRate := int32(16000)
+	samplesPer10ms := sampleRate / 100
+	channel := int32(1)
+	bytesPerSample := int32(2)
+	pcmFrameSize := int(samplesPer10ms * channel * bytesPerSample) // per 10ms
+
+	// init chan
+	textChan = make(chan *message, textChanMax)
+
+	go func() {
+		slog.Info("process textChan", logTag)
+
+		for msg := range textChan {
+			if msg.receivedTs < outdateTs.Load() { // Check whether to interrupt
+				slog.Info(fmt.Sprintf("textChan interrupt and flushing for input text: [%s], receivedTs: %d, outdateTs: %d",
+					msg.text, msg.receivedTs, outdateTs.Load()), logTag)
+				continue
+			}
+
+			wg.Add(1)
+			slog.Info(fmt.Sprintf("textChan text: [%s]", msg.text), logTag)
+
+			r, w := io.Pipe()
+
+			go func() {
+				defer wg.Done()
+				defer w.Close()
+
+				slog.Info(fmt.Sprintf("textToSpeechStream text: [%s]", msg.text), logTag)
+
+				err = e.elevenlabsTTS.textToSpeechStream(w, msg.text)
+				if err != nil {
+					slog.Error(fmt.Sprintf("textToSpeechStream failed, err: %v", err), logTag)
+					return
+				}
+			}()
+
+			slog.Info(fmt.Sprintf("read PCM stream, text:[%s], pcmFrameSize:%d", msg.text, pcmFrameSize), logTag)
+
+			var n int
+			buf := make([]byte, pcmFrameSize)
+
+			// read PCM stream
+			for {
+				if msg.receivedTs < outdateTs.Load() { // Check whether to interrupt
+					slog.Info(fmt.Sprintf("read PCM stream interrupt and flushing for input text: [%s], receivedTs: %d, outdateTs: %d",
+						msg.text, msg.receivedTs, outdateTs.Load()), logTag)
+					break
+				}
+
+				n, err = r.Read(buf)
+
+				if err == io.EOF {
+					break
+				}
+				if err != nil {
+					slog.Error(fmt.Sprintf("read PCM stream failed, err: %v", err), logTag)
+					break
+				}
+
+				if n != pcmFrameSize {
+					slog.Warn(fmt.Sprintf("the number of bytes read is [%d] inconsistent with PCM frame size", n), logTag)
+					continue
```

Comment:
These data still could be used, just fill in blank for the rest. It's common at the end.